clang/lib/Headers/amdgpuintrin.h

```diff
@@ -13,11 +13,8 @@
 #error "This file is intended for AMDGPU targets or offloading to AMDGPU"
 #endif
 
-#include <stdint.h>
-
-#if !defined(__cplusplus)
-_Pragma("push_macro(\"bool\")");
-#define bool _Bool
+#ifndef __GPUINTRIN_H
+#warning "This file is intended as an implementation detail of gpuintrin.h"
```

Comment:
We should do this like the SSE headers and do something like
```suggestion
#error "Never use <amdgpuintrin.h> directly; include <gpuintrin.h> instead."
```

Reply:
yep, done